ginisdk/src/androidTest/java/net/gini/android/SdkIntegrationTestAccounting.java

```diff
@@ -0,0 +1,361 @@
+        final DocumentUploadBuilder uploadBuilder = new DocumentUploadBuilder().setDocumentBytes(testDocument).setDocumentType(DocumentTaskManager.DocumentType.INVOICE);
+        final DocumentTaskManager documentTaskManager = gini.getDocumentTaskManager();
+
+        final Task<Document> upload = uploadBuilder.upload(documentTaskManager);
+        final Task<Document> processDocument = upload.onSuccessTask(new Continuation<Document, Task<Document>>() {
+            @Override
+            public Task<Document> then(Task<Document> task) throws Exception {
+                Document document = task.getResult();
+                return documentTaskManager.pollDocument(document);
+            }
+        });
+
+        final Task<Map<String, SpecificExtraction>> retrieveExtractions = processDocument.onSuccessTask(new Continuation<Document, Task<Map<String, SpecificExtraction>>>() {
+            @Override
+            public Task<Map<String, SpecificExtraction>> then(Task<Document> task) throws Exception {
+                return documentTaskManager.getExtractions(task.getResult());
+            }
+        });
+
+        retrieveExtractions.waitForCompletion();
+        if (retrieveExtractions.isFaulted()) {
+            Log.e("TEST", Log.getStackTraceString(retrieveExtractions.getError()));
+        }
+
+        assertTrue("extractions shouldn't have succeeded", retrieveExtractions.isFaulted());
+    }
+
+    @SdkSuppress(maxSdkVersion = Build.VERSION_CODES.JELLY_BEAN_MR1)
+    public void testPublicKeyPinningWithMultiplePublicKeys() throws Exception {
+        resetTrustKit();
+        gini = new SdkBuilder(getContext(), clientId, clientSecret, "example.com").
+                setNetworkSecurityConfigResId(net.gini.android.test.R.xml.multiple_keys_network_security_config).
+                setApiBaseUrl(apiUriAccounting).
+                setGiniApiType(GiniApiType.ACCOUNTING).
+                setUserCenterApiBaseUrl(userCenterUri).
+                setConnectionTimeoutInMs(60000).
+                build();
+
+        final AssetManager assetManager = getContext().getResources().getAssets();
+        final InputStream testDocumentAsStream = assetManager.open("test.jpg");
+        assertNotNull("test image test.jpg could not be loaded", testDocumentAsStream);
+
+        final byte[] testDocument = TestUtils.createByteArray(testDocumentAsStream);
+        final DocumentUploadBuilder uploadBuilder = new DocumentUploadBuilder().setDocumentBytes(testDocument).setDocumentType(DocumentTaskManager.DocumentType.INVOICE);
+        processDocument(uploadBuilder);
+    }
+
+    private String extractEmailDomain(String email) {
+        String[] components = email.split("@");
+        if (components.length > 1) {
+            return components[1];
+        }
+        return "";
+    }
+
+    private Map<Document, Map<String, SpecificExtraction>> processDocument(DocumentUploadBuilder uploadBuilder) throws InterruptedException, JSONException {
```

Comment:
I'm not super convinced about this method name, since it is called in every test to assert some result. Maybe `assertValidExtractions`?